### ISSUE TITLE
Returning complete results of scanned trie for a sentiment analysis.

### ISSUE THREAD

**Issue by ezrabutler** [spencermountain/compromise#807]
Hi,

I've created a super quick sentiment analyzer by building a trie with the [AFINN 165](https://github.com/words/afinn-165), scanning the trie with my document, and then iterating through the results to see if there are any words with a negative valence. It worked well, except in cases where it's a two-word phrase in AFINN (eg "F****** Amazing" spelled out is a +4, whereas the curse word alone is a -4. )

So I mapped two lists of words by valence (+ or -), and if then I filtered out the intersection from the negative list. This all works great, and "F****** Amazing" will only return the positive sentiment and not the negative one.

```
let trie = nlp.buildTrie(AFINN)
let doc = nlp(request.text)
const scanned = doc.scan(trie)
var positive = [];
var negative = [];
  for (var prop in scanned){
    if (prop > 0) {positive =  scanned[prop].map(list => list.list).map(word => word[0].start)}
    if (prop < 0) {negative =  scanned[prop].map(list => list.list).map(word => word[0].start)}
  if (negative.length && positive.length) {let intersection = negative.filter(x => positive.includes(x));
  if (intersection.length > 0) negative = negative.filter(x => !intersection.includes(x));
  
}
if (negative.length > 0){window.negativity.push(negative)}  
} 
```

The problem happens when I'm trying to use a positive phrase like "the sh**", which returns a positive sentiment from the two words together, but a negative one from having scanned the second word. The problem seems to be that there is no "simple" way to return the entire phrase (or two add both words from the phrase) to the positive sentiment. 

As I'm trying to train this to only identify negatives, it is not working. (I realize that this is an edge case which is probably very unlikely.) 

In short, the trie scan works perfectly, but I'm getting a false "negative" in cases where the second word of a phrase is on both lists, and I can't figure out how to include the entire phrase when returning the results of the scanned trie. 

This is the stringified result. Even if I were to look at the list[0].length, I can't figure out how to access the .next, as it's not included in the results. Am I able to modify what the returned object is?
```
{"4":{"list":[{"start":"the-84odgqA","length":2,"isA":"Phrase"}]},"-4":{"list":[{"start":"s**t-f6jD8yQ","length":1,"isA":"Phrase"}]}}

```

Any help would be appreciated.

Thank you.
Ezra

**Comment by spencermountain** [comment]
hey Ezra, sounds like a cool project. Thanks for sharing that AFinn list.

Yeah, this should be possible - two ideas.

1 -  try adding your false-positive matches eg 'the shit' to the trie aswell - (before the shorter versions!) Then it will match those before the shorter ones.

2 - .scan() returns a regular compromise document, which you can jump-around in most directions. You may find use for `.parent()` or `.sentence()` or [.lookAhead() .lookBehind()](https://observablehq.com/@spencermountain/compromise-match) for doing an additional lookup on the scanned result

let me know how it goes!
cheers

**Comment by spencermountain** [comment]
https://runkit.com/spencermountain/6018a1f2a07a61001ab8ae21

**Comment by ezrabutler** [comment]
Thank you so much!

I put "the shit" first in the list - I didn't realize that the trie parses in order. But it still returns the same response -- ie tagging "shit" in addition to "the shit". Would love to be able to force a single pass, but I guess not. Ultimately it isn't the end of the world. 
I'll play around more with the match() capabilities with the lookAhead() and lookBehind(). And thank you for the example. 

Your whole project is insanely cool, so thank you.
